pennylane/ops/meta.py

```diff
@@ -238,7 +238,7 @@ def __init__(
             measurement = qml.state()
         if isinstance(measurement, qml.measurements.StateMP) and shots == "workflow":
             shots = None  # always use analytic with state
-        if isinstance(measurement, qml.ops.MidMeasure):
+        if isinstance(measurement, (qml.ops.MidMeasure, qml.ops.PauliMeasure)):
             raise ValueError("Mid-circuit measurements can not be used in snapshots.")
```

Comment:
Actually, we should be able to get rid of this now? Since it will error out on the next line because they are not measurements anymore.

Reply:
Good point.